src/doc/components/Esdoc/SummaryDoc.js

```diff
@@ -77,17 +82,24 @@ const SummaryDoc = ({
                     )}
                     <span data-ice="async">{doc.async ? 'async' : ''}</span>
                     <span data-ice="generator">{doc.generator ? '*' : ''}</span>
-                    <span className="code" data-ice="name">
-                      <DocLinkHTML
-                        longname={doc.longname}
-                        text={null}
-                        inner={innerLink}
-                        kind={doc.kind}
-                      />
-                    </span>
-                    <span className="code" data-ice="signature">
-                      <SignatureHTML doc={doc} />
-                    </span>
+                    {/* Custom return if 'constructor' */}
+                    {doc.kind === 'constructor' ? (
+                      `new ${doc.longname.match(/~([^)]+)#constructor/)[1]}()`
+                    ) : (
+                      <>
+                        <span className="code" data-ice="name">
+                          <DocLinkHTML
+                            longname={doc.longname}
```

Comment:
only this attribute  change if it's a constructor

Reply:
there is already a condition
`doc.kind === 'constructor'`